process/transaction/shardProcess.go

```diff
@@ -293,16 +293,18 @@ func (txProc *txProcessor) createReceiptWithReturnedGas(
 	tx *transaction.Transaction,
 	acntSnd state.UserAccountHandler,
 	moveBalanceCost *big.Int,
+	totalProvided *big.Int,
 	destShardTxType process.TransactionType,
+	isUserTxOfRelayed bool,
 ) error {
-	if check.IfNil(acntSnd) {
+	if check.IfNil(acntSnd) || isUserTxOfRelayed {
 		return nil
 	}
-	if destShardTxType != process.MoveBalance {
+	if destShardTxType != process.MoveBalance ||
```

Comment:
maybe add a bool value for the if condition?

Reply:
the second part is for backward compatibility.